src/openfermion/measurements/_time_series_estimators.py

```diff
@@ -0,0 +1,460 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Classes for time series estimators for quantum phase estimation.
+Estimators described in arXiv:1809.09697, in particular in Sec.II.A.
+"""
+
+import numpy as np
+import scipy as sp
+
+
+class TimeSeriesEstimator(object):
+    '''
+    Time-series estimator: a QPE estimator based on a time-series
+        analysis of g(k)=sum_jA_je^{i*k*phi_j}, where
+        A_j=|<phi_j|Psi>|^2 is the probabilistic overlap
+        with the starting state |Psi>, and phi_j is the unitary
+        eigenvalue U|phi_j>=e^{i*phi_j}|phi_j>.
+    '''
+    def __init__(self,
+                 max_experiment_length,
+                 num_freqs_max,
+                 depolarizing_noise=False,
+                 singular_values_cutoff=1e-12,
+                 store_history=False,
+                 automatic_average=False):
+        '''
+        Args:
+            max_experiment_length: the largest number (K) of unitary operations
```

Comment:
Document the type of the arguments in parentheses; for example, `max_experiment_length (int)`. This applies to the rest of the PR too.

Reply:
I think I've got all of these.